share/wake/lib/system/job.wake

```diff
@@ -543,6 +543,10 @@ export def getPathParent path = match path
   Path name = Path (simplify "{name}/..")
   BadPath e = BadPath e
 
+export def isPathDir = match _
+  BadPath _ = False
+  Path str = matches `0+` (hashcode str)
```

Comment:
Is `hashcode` expensive for large files?

Reply:
It's free. This is already in the database from when the Path was created, so it's cheaper than stat, and avoids the race condition.